### PR TITLE
Create JoinRoom component

### PR DESCRIPTION
**Pull request by alexrkoch**
This change creates a small component to handle joining an existing room.

It accepts a regex as input which will be the room-code's format, allowing the user to join when a valid code has been entered.

### CODE REVIEW

**Comment by alexrkoch** [comment]
Current UI (still un-styled):
Screen Shot 2023-06-09 at 6 24 27 PM

File: src/routes/+page.svelte [hunk 1]
```diff
@@ -1,9 +1,12 @@
-<script>
-	import CreateRoom from "$lib/CreateRoom.svelte";
+<script lang="ts">
+	import CreateRoom from '$lib/CreateRoom.svelte';
+	import JoinRoom from '$lib/JoinRoom.svelte';
 
-  let buttonText = "Create a New Room"
+	let buttonText: string = 'Create a New Room';
+	let roomCodePattern: RegExp = /^[A-Z]{6}$/;
```
Comment (mitchpowell1) [inline src/routes/+page.svelte]: nit: Been a while since I've javascripted, but can these be const?

File: src/lib/JoinRoom.svelte [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<script lang="ts">
+	export let roomCodePattern: RegExp = /.*/; // default to match anything
```
Comment (mitchpowell1) [inline src/lib/JoinRoom.svelte:2]: Looks like there's maybe some duplication of the room code pattern matching between this file and the other one, could they be unified?

**Review by mitchpowell1 — APPROVED** [review]

File: src/routes/+page.svelte (hunk 1 above)
Comment (alexrkoch) [reply src/routes/+page.svelte]: nah you right, I will const these bad boyes up

File: src/lib/JoinRoom.svelte (hunk 2 above)
Comment (alexrkoch) [reply src/lib/JoinRoom.svelte:2]: I think both are needed- The "export .." declaration inside the component basically defines what can be passed in as props. Then the declaraction in the "parent" component defines the variable they want to pass in

**Pull request merged**